packages/dds/shared-object-base/src/serializer.ts

```diff
@@ -19,6 +19,22 @@ import {
 	RemoteFluidObjectHandle,
 } from "@fluidframework/runtime-utils/internal";
 
+/**
+ * Functionality needed to bind a handle to itself such that the target handle is "as attached as" this source object
+ *
+ * @internal
+ */
+export interface HandleBinder {
+	bind(handle: IFluidHandleInternal): void;
+}
+
+/**
+ * Type guard for {@link HandleBinder}
+ */
+function supportsBind(handle: unknown): handle is HandleBinder {
+	return typeof (handle as HandleBinder).bind === "function";
```

Comment:
this needs to be something better, all native javascript methods have a bind call, so lots of object fit here.

Reply:
Good point! That is funny and not funny.

Reply:
one options might be to just do `instanceof SharedObjectHandle`. that is safe within the package, and they you don't need another interface either. the concreteness will always make the flow easier to understand.

Reply:
it looks like SharedObjectHandle isn't exported, so maybe add an interface for SharedObjectHandle which can be exported.

Reply:
either way, lets make the typing concrete to keep it easy to understand that only dds bind

Reply:
I started with `instanceof SharedObjectHandle` (see initial commit) but there's a tree test that plugs in a dummy object so I went with this.